src/Simulator/IncrementalRedstoneSimulator.cpp

```diff
@@ -503,152 +514,280 @@ void cIncrementalRedstoneSimulator::HandleRedstoneButton(int a_RelBlockX, int a_
 
 void cIncrementalRedstoneSimulator::HandleRedstoneWire(int a_RelBlockX, int a_RelBlockY, int a_RelBlockZ)
```

Comment:
This method is too big, so if you're going to make significant changes, split it up